website/source/docs/providers/aws/r/lambda_permission.html.markdown

```diff
@@ -27,14 +27,14 @@ resource "aws_lambda_permission" "allow_cloudwatch" {
 resource "aws_lambda_alias" "test_alias" {
   name             = "testalias"
   description      = "a sample description"
-  function_name    = "${aws_lambda_function.test_lambda.arn}"
+  function_name    = "${aws_lambda_function.test_lambda.function_name}"
   function_version = "$LATEST"
 }
 
 resource "aws_lambda_function" "test_lambda" {
   filename      = "lambdatest.zip"
   function_name = "lambda_function_name"
-  role          = "${aws_iam_role.iam_for_lambda.arn}"
+  role          = "${aws_iam_role.iam_for_lambda.name}"
```

Comment:
I agree the attribute name may be confusing, however the API in fact rejects names and only accepts ARNs, see what happens when you try to apply the changed example:

```
$ terraform apply
aws_iam_role.iam_for_lambda: Creating...
  arn:                "" => "<computed>"
  assume_role_policy: "" => "{\n  \"Version\": \"2012-10-17\",\n  \"Statement\": [\n    {\n      \"Action\": \"sts:AssumeRole\",\n      \"Principal\": {\n        \"Service\": \"lambda.amazonaws.com\"\n      },\n      \"Effect\": \"Allow\",\n      \"Sid\": \"\"\n    }\n  ]\n}\n"
  create_date:        "" => "<computed>"
  name:               "" => "iam_for_lambda_rsimko"
  path:               "" => "/"
  unique_id:          "" => "<computed>"
aws_iam_role.iam_for_lambda: Creation complete (ID: iam_for_l...a_rsimko)
aws_lambda_function.test_lambda: Creating...
  arn:              "" => "<computed>"
  filename:         "" => "../lambdatest.zip"
  function_name:    "" => "lambda_function_name_rsimko"
  handler:          "" => "exports.handler"
  last_modified:    "" => "<computed>"
  memory_size:      "" => "128"
  publish:          "" => "false"
  qualified_arn:    "" => "<computed>"
  role:             "" => "iam_for_lambda_rsimko"
  runtime:          "" => "nodejs4.3"
  source_code_hash: "" => "<computed>"
  timeout:          "" => "3"
  version:          "" => "<computed>"
Error applying plan:

1 error(s) occurred:

* aws_lambda_function.test_lambda: 1 error(s) occurred:

* aws_lambda_function.test_lambda: Error creating Lambda function: ValidationException: 1 validation error detected: Value 'iam_for_lambda_rsimko' at 'role' failed to satisfy constraint: Member must satisfy regular expression pattern: arn:aws:iam::\d{12}:role/?[a-zA-Z_0-9+=,.@\-_/]+
	status code: 400, request id: f37875c2-1516-11e7-8be2-a1a8bd2483c5

```